R/g.sib.det.R

```diff
@@ -195,12 +195,27 @@ g.sib.det = function(M, IMP, I, twd = c(-12, 12),
         firstmidnight = midnights[1]
         firstmidnighti = midnightsi[1]
       }
-      for (j in 1:(countmidn)) { #Looping over the midnight
-        qqq1 = midnightsi[j] + (twd[1] * (3600 / ws3)) #preceding noon
-        qqq2 = midnightsi[j] + (twd[2] * (3600 / ws3)) #next noon
+      # if recording started before 4am, then also derive first awakening
+      first4am = grep("04:00:00", time)[1]
+      if (first4am < firstmidnighti) { # this means recording started after midnight and before 4am
+        midn_start = 0
+      } else {
+        midn_start = 1
+      }
+      sptei = 0
+      for (j in midn_start:(countmidn)) { #Looping over the midnight
+        if (j == 0) {
+          qqq1 = 1 # preceding noon (not available in recording)
+          qqq2 = grep("12:00:00", time)[1] # first noon in recording
+          sptei = 1
+        } else {
+          qqq1 = midnightsi[j] + (twd[1] * (3600 / ws3)) #preceding noon
+          qqq2 = midnightsi[j] + (twd[2] * (3600 / ws3)) #next noon
+          sptei = sptei + 1
```

Comment:
Simplification suggestion:
Move  `sptei = sptei + 1` to after line 215, such that line 210 and 214 can be removed.